src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/MidEventCallback.java

```diff
@@ -77,6 +85,18 @@ public JsonNode invoke(String caseTypeId,
         return dataJsonNode(content.getData());
     }
 
+    private Optional<CaseDetails> getCaseDetails(CaseDataContent content, CaseType caseType) {
+        try {
+            CaseDetails caseDetails = caseService.getCaseDetails(caseType.getJurisdictionId(), content.getCaseReference());
```

Comment:
Do we need to call this method if case reference is null? CaseService.getCaseDetails also throws BadRequestException if the case reference is invalid.